### PR TITLE
perf(contract): optimize `StreamRegistry` again

### PR DESCRIPTION
**Pull request by shuhuiluo**
- Add tests for stream-node mapping and validate edge cases
- Refactor stream validation with reusable internal functions

### CODE REVIEW

File: contracts/src/river/registry/facets/stream/StreamRegistry.sol [hunk 1]
```diff
@@ -289,16 +280,29 @@ contract StreamRegistry is IStreamRegistry, RegistryModifiers {
...
+    //    for (uint256 i; i < streamLength; ++i) {
+    //      bytes32 id = ds.streams.at(i);
+    //      Stream storage stream = ds.streamById[id];
+    //      for (uint256 j; j < stream.nodes.length; ++j) {
+    //        if (stream.nodes[j] == nodeAddress) {
+    //          ++count;
+    //          break;
+    //        }
+    //      }
+    //    }
```
Comment (texuf) [inline contracts/src/river/registry/facets/stream/StreamRegistry.sol:294]: Can we delete the commented out code?

Comment (shuhuiluo) [reply contracts/src/river/registry/facets/stream/StreamRegistry.sol:294]: That's the intention. In fact the current code will run out of gas anyway.

**Comment by shuhuiluo** [comment]
Moving new storage variables to a new PR.

**Review by giuseppecrj — APPROVED** [review]
